torch/csrc/jit/passes/onnx/peephole.cpp

```diff
@@ -287,9 +287,97 @@ void removeNopPacking(std::shared_ptr<Graph>& graph) {
 
     n->removeAllInputs();
     it.destroyCurrent();
+
+    if (input->outputs()[0]->uses().size() == 0) {
+      input->destroy();
+    }
+  }
+}
+
+void fixDefaultRNNState(std::shared_ptr<Graph>& graph, Node * n, int input_index) {
+  auto initial_state = n->inputs()[input_index];
+  if (initial_state->node()->kind() == onnx::Constant ||
+      (initial_state->node()->kind() == onnx::Slice &&
+       initial_state->node()->inputs()[0]->node()->kind() == onnx::Constant)) {
+    Node * shape_of_input = graph->create(onnx::Shape, 1);
+    shape_of_input->insertBefore(n);
+    shape_of_input->addInput(n->inputs()[0]);
+
+    Node * gather_indices = graph->create(onnx::Constant, 1);
+    gather_indices->insertBefore(n);
+    gather_indices->t_(attr::value, at::Scalar(1).toTensor());
+
+    Node * batch_size = graph->create(onnx::Gather, 1);
+    batch_size->insertBefore(n);
+    batch_size->addInput(shape_of_input->outputs()[0]);
+    batch_size->addInput(gather_indices->outputs()[0]);
+
+    Node * unsqueezed_batch_size = graph->create(onnx::Unsqueeze, 1);
+    unsqueezed_batch_size->insertBefore(n);
+    unsqueezed_batch_size->addInput(batch_size->outputs()[0]);
+    unsqueezed_batch_size->is_(attr::axes, {0});
+
+    Node * hidden_size = graph->create(onnx::Constant, 1);
+    hidden_size->insertBefore(n);
+    hidden_size->t_(attr::value, at::Scalar(n->i(attr::hidden_size)).toTensor());
+
+    Node * unsqueezed_hidden_size = graph->create(onnx::Unsqueeze, 1);
+    unsqueezed_hidden_size->insertBefore(n);
+    unsqueezed_hidden_size->addInput(hidden_size->outputs()[0]);
+    unsqueezed_hidden_size->is_(attr::axes, {0});
```

Comment:
Couldn't we set `hidden_size` to be a 1d tensor immediately, and avoid having those unsqueezes?

Reply:
possibly, I'll give it a try